dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/BaseControls/Mechanical.mo

```diff
@@ -0,0 +1,103 @@
+    Dialog(tab = "Mechanical"));
+
+  //Input variables
+  Modelica.Blocks.Interfaces.RealInput PAeroPu(start = PAg0Pu) "Aerodynamic power in pu (base SNom) (generator convention)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 54}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput PAgPu(start = PAg0Pu) "Generator (air gap) power in pu (base SNom) (generator convention)" annotation(
+    Placement(visible = true, transformation(origin = {-120, -54}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, -40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  //Output variables
+  Modelica.Blocks.Interfaces.RealOutput omegaWTRPu(start = SystemBase.omega0Pu) "Rotor angular frequency in pu (base omegaNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput omegaGenPu(start = SystemBase.omega0Pu) "Generator angular frequency in pu (base omegaNom)" annotation(
+    Placement(visible = true, transformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Modelica.Blocks.Math.Add add(k2 = -1)  annotation(
+    Placement(visible = true, transformation(origin = {-10, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add add1(k2 = -1)  annotation(
+    Placement(visible = true, transformation(origin = {-10, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.Integrator integrator(k = 1 / (2 * Hwtr), y_start = SystemBase.omega0Pu)  annotation(
+    Placement(visible = true, transformation(origin = {30, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.Integrator integrator1(k = 1 / (2 * Hgen), y_start = SystemBase.omega0Pu)  annotation(
+    Placement(visible = true, transformation(origin = {30, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Division division annotation(
+    Placement(visible = true, transformation(origin = {-70, 60}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+  Modelica.Blocks.Math.Division division1 annotation(
+    Placement(visible = true, transformation(origin = {-70, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add add2(k2 = -1)  annotation(
+    Placement(visible = true, transformation(origin = {30, 0}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
+  Dynawo.NonElectrical.Blocks.Continuous.PI pI(Ki = KdrtPu, Kp = CdrtPu, Y0 = PAg0Pu / SystemBase.omegaRef0Pu)  annotation(
```

Comment:
omega0Pu

Reply:
Also, PAg0Pu or P0Pu * SystemBase.SnRef / SNom ?

Reply:
Doesn't matter, you choose one, you don't really have a choice. (as we said omega starts at omega0Pu and not PAg0Pu / P0Pu (or reverse) we can't decide